src/libstore/derivations.cc

```diff
@@ -272,7 +272,15 @@ Derivation parseDerivation(const Store & store, std::string && s, std::string_vi
 
 static void printString(string & res, std::string_view s)
 {
-    char buf[s.size() * 2 + 2];
+    char * buf;
+    size_t bufSize = s.size() * 2 + 2;
+    std::unique_ptr<char[]> dynBuf;
+    if (bufSize < 0x10000) {
+        buf = (char *)alloca(bufSize);
```

Comment:
FYI, alloca is not standard C/POSIX, and definitely not idiomatic C++.

Reply:
Ok, I can make it always allocate on the heap if you prefer, but that's going to be the obvious solution in case Nix is ported to such a rare platform.

Reply:
you could also move the actual buffer using code into a local lambda and either pass a stack-allocated buffer or a dynamically allocated buffer. avoiding heap allocs in this part of the code is worth it